File: src/client/classes/api/req_res/add-kc-response.ts
'use strict'

import { KC } from '@/classes/datas/kc'
import { GkillAPIResponse } from '../gkill-api-response'
import { Kyou } from '@/classes/datas/kyou'

export class AddKCResponse extends GkillAPIResponse {

    added_kc: KC

    added_kyou: Kyou | null

    constructor() {
        super()
        this.added_kc = new KC()
        this.added_kyou = null
    }

}

File: src/server/gkill/api/req_res/add_kc_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/server/gkill/api/message"
	"github.com/mt3hr/gkill/src/server/gkill/dao/reps"
)

type AddKCResponse struct {
	Messages message.GkillMessages `json:"messages"`

	Errors message.GkillErrors `json:"errors"`

	AddedKC *reps.KC `json:"added_kc"`

	AddedKyou *reps.Kyou `json:"added_kyou"`
}
